### PR TITLE
As part of the swagger AuthN/AuthZ changes:

### PR DESCRIPTION
**Pull request by zyoung51**
- Add local user model and use bcrypt to serialize passwords
- Add roles model
- Add 401 and 403 errors to the common errors

### CODE REVIEW

**Comment by anhou** [comment]
:+1:

**Comment by zyoung51** [comment]
test this please

File: lib/models/local-user.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                required: true,
+                minLength: 6,
+                maxLength: 50
+            },
+            role: {
+                type: 'string',
+                defaultsTo: 'ReadOnly'  // change to Constants
+            },
+            comparePassword: function (password) {
+                var combined = new Buffer(this.password, 'base64');
```
Comment (WangWinson) [inline lib/models/local-user.js]: The compare password maybe a common function not limit to user model ? For example, we need to hash BMC password, too. So we may extract it as a common function later. This is nice to have.

**Comment by WangWinson** [comment]
:+1:

File: lib/models/local-user.js (hunk 1 above)
Comment (zyoung51) [inline lib/models/local-user.js]: This compare password is using a one way hash function where-as the BMC passwords need to be support decryption so they can be used by the RackHD pollers, etc.

**Comment by jlongever** [comment]
+1

**Pull request merged**